src/HYPRE.jl

```diff
@@ -344,36 +343,49 @@ function subarray_unsafe_supported()
     return VERSION >= v"1.9.0"
 end
 
-# TODO: This has some duplicated code with to_hypre_data(::SparseMatrixCSC, ilower, iupper)
-function Internals.to_hypre_data(A::SparseMatrixCSC, r::AbstractLocalIndices, c::AbstractLocalIndices)
-    g_to_l_rows = global_to_local(r) # Not sure about this assert
-    l_to_g_rows = local_to_global(r)
-    @assert g_to_l_rows.own_to_local isa UnitRange && g_to_l_rows.own_to_local.start == 1
-
-    n_local_rows = own_length(r)
-    n_local_cols = own_length(c)
-    ilower = l_to_g_rows[1]
-    iupper = l_to_g_rows[own_length(r)]
-    a_rows = rowvals(A)
-    a_vals = nonzeros(A)
+function Internals.to_hypre_data(
+        A::SplitMatrix{<:SparseMatrixCSC},
+        r::AbstractLocalIndices, c::AbstractLocalIndices
+    )
+    # Own/ghost to global index mappings
+    own_to_global_row = own_to_global(r)
+    own_to_global_col = own_to_global(c)
+    ghost_to_global_col = ghost_to_global(c)
+
+    # HYPRE requires contiguous row indices
+    ilower = own_to_global_row[1]
+    iupper = own_to_global_row[end]
+    @assert iupper - ilower + 1 == length(own_to_global_row)
+
+    # Extract sparse matrices from the SplitMatrix. We are only interested in the owned
+    # rows, so only consider own-own and own-ghost blocks.
+    Aoo = A.blocks.own_own::SparseMatrixCSC
```

Comment:
@fverdugo is using `A.block.own_own` the correct way or is there some function to extract this part? I saw https://github.com/fverdugo/PartitionedArrays.jl/blob/b211949cc5842f3d6d99f8452b9f0233aef963a5/src/p_sparse_matrix.jl#L670-L672 but not sure what to pass as the second and third argument (although they are unused so doesn't matter...)